Oracle.md

```diff
@@ -0,0 +1,128 @@
+# Oracle specifications
+
+## Oracle templates
+
+An oracle template provides information to clients about an event for which an oracle plans on releasing a signature over its outcome.
+The provided information should be sufficient for a client to create a set of CETs that will cover all possible outcomes of the event.
+
+Regardless of the outcome type, an oracle template should include:
+
+- The R value that will be used to sign the event outcome (possibly several).
+- The public key of the oracle.
+- The earliest date at which the event outcome will be released.
+
+Here we assume that an event outcome can be represented either as a set of strings or numbers.
+Three kinds of event outcomes are defined (note that a single even can be composed of several types).
+
+### Simple enumeration
+
+For events that have a narrow range of possible outcomes, the possible outcomes can simply be enumerated.
+For the sake of simplicity, outcomes should be represented as strings.
+
+#### Example: Weather tomorrow
+
+Tomorrow's weather can be represented as the set of strings `[sunny, cloudy, rainy]`.
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- end: the last possible outcome number
+- step: the increment
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+end: 100
+step: 1
+```
+
+### Large range
+
+When a range of a numerical outcome is sufficiently large or unbounded, the oracle can represent the outcome using exponent/mantissa (using true normalized form).
+
+The oracle template should include:
+
+- base: the base in which the outcome is represented
+
+The oracle will then release two R value `R_exponent` and `R_mantissa` as well as two signatures `s_exponent` and `s_mantissa`.
+
+#### Example: BTC/USD rate
+
+```
+base: 10
+exponent: 5
+mantissa: 0.9
+```
+
+### Serialization and signature
+
+Oracle templates should be serialized using TLV (to be defined).
+The oracle should provide a signature over the serialized template to the client as an attestation that he committed to release a signature over the event outcome.
```

Comment:
Good question, not sure if it matters so much here since this is mainly for accountability, maybe @LLFourn has some argument in favor of one or another?

Reply:
use the same signature scheme as the anticipated signature scheme i.e. bip340 schnorr

Reply:
Specified that it should follow BIP 340